Documentation update: Concepts (#2218)
fetchai/agents-aea: docs/language-agnostic-definition.md

Currently, there is an implementation of the AEA framework in Python which enables the development of AEAs in Python, and allows AEAs which are built with it to run.

> However, AEAs, by design, are not constrained by any technical requirement, including the choice of programming language. This is further backed by the idea that agent-based solutions are suited for multi-stakeholder environments where the different AEAs may be developed independently of one another, resulting in a heterogeneous system. 

Reviewer: ```suggestion
However, AEAs can be developed in many different programming languages. This is further backed by the idea that agent-based solutions are suited for multi-stakeholder environments where the different AEAs may be developed independently of one another, resulting in a heterogeneous system. 
```

Reviewer: There are technical constraints of course.

Author: Done

This means that in principle, there could be different implementations of the AEA framework, in various programming languages and for different platforms. However, to ensure that AEAs under any implementation are compatible with one another and able to interact, they must satisfy specific definitions. In this page, we compile a set of definitions which any AEA independent of its implementation must satisfy in order to be able to interact with other AEAs.

> An Autonomous Economic Agent, in technical terms, must satisfy the following requirements:

Reviewer: ```suggestion
An AEA, in technical terms, must satisfy the following requirements:
```

Author: Done

<ul>

> <li> It MUST be capable of receiving and sending <code>Envelopes</code> with the following <a href="https://developers.google.com/protocol-buffers" target="_blank">protobuf</a> schema:

Reviewer: please revert, the envelope is not sent or received *with* the schema, the schema defines how it is encoded/decoded.

Author: Done
